tests/integration/defs/disaggregated/test_aiperf_gate.py

```
@@ -0,0 +1,152 @@
+    with pytest.raises(AssertionError, match="exceeds threshold"):
+        enforce_aiperf_error_rate(artifact_dir, 0.05, expected_records=35000)
+
+
+def test_passes_healthy_run_with_cancellations(tmp_path):
+    records = [_record()] * 898 + [_record(_CANCEL)] * 100 + [_record(_SERVER_500)] * 2
+    artifact_dir = _write_export(tmp_path, records)
+    enforce_aiperf_error_rate(artifact_dir, 0.05, expected_records=1000)
+
+
+def test_missing_export_raises(tmp_path):
+    with pytest.raises(FileNotFoundError, match="profile_export.jsonl"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)